doc/Reactive.md

```diff
@@ -1,13 +1,67 @@
+	}
+}
+```
+
+But in both cases, if there is any exception the stream will be broken. This means that for instance in example above, if we cannot compute the shipping cost for a given country for any reason (network issue, invalid country, …) the stream of data will be terminated, and selecting another country won’t have any effect.
+
+Also, when a dependency is being updated and we may need to do some asynchronously work, like update a projection. In our example, we asynchronously get the updated shipping cost when country is changed. From a UI perspective, it would be great to have a visual indication that the shipping cost is being re-computed for the newly chosen country.
+
+Neither `IObservable<T>` nor `IAsyncEnumerable<T>` have such metadata mechanism for produced values, that the purpose if `IFeed<T>`.
+
+With the data, currently `IFeed<T>` does supports 2 mains metadata (named “axis”):
+•	Error: If there is any exception linked to the current data
+•	Progress: We indicates that the current data is transient or final.
+But we do also have a metadata about the data itself:
+•	Some: The _feed_ does have a valid data.
+•	None: There is no data to display. In our example, when you cannot ship to the selected country.
+•	Undefined: There is no info about the data, typically because we are asynchronously loading it.
+
+Here a diagram of the common messages produced by a feed when asynchronously loading a data:
+
+
+> Keep in mind that this is only an example of the common case, but each _axis_ is independent and can change from one state to another. There is no restriction between states.
+
+
+
 
-This package is a building block to create a [reactive application](https://en.wikipedia.org/wiki/Reactive_programming).
 
 ## API
 The main API provided by this package is [`IFeed<T>`](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive/Core/IFeed.cs) which represents a stream of _data_.
 
-Unlike `IObservable<T>` or `IAsyncEnumerable<T>`, the _feed_ streams are specialized to handle business data objects which are expected to be rendered by the UI. A _feed_ is a sequence of _messages_ that contains such immutable _data_ with some metadata like its loading progress and/or errors raised while loading it. Everything is about the _data_, this means that a _feed_ will never fail, it will instead report any error encountered with the data itself, remaining active for future updates.
+Unlike `IObservable<T>` or `IAsyncEnumerable<T>`, the _feed_ streams are specialized to handle business data objects which are expected to be rendered by the UI.
+A _feed_ is a sequence of _messages_ that contains such immutable _data_ with some metadata like its loading progress and/or errors raised while loading it.
+Everything is about the _data_, this means that a _feed_ will never fail, it will instead report any error encountered with the data itself, remaining active for future updates.
```

Comment:
```suggestion
Everything is about _data_, this means that a _feed_ will never fail. It will instead report any error encountered with the data itself, remaining active for future updates.
```